clusters/dev/dns.tf

```diff
@@ -1,5 +1,27 @@
 resource "google_dns_managed_zone" "root" {
   name        = "mindtastic-zone-${var.environment}"
-  dns_name    = "${var.environment}.mindtastic.lol."
+  dns_name    = var.cluster_dns_zone_name
   description = "The base DNS zone for ${var.environment} environment"
-}
\ No newline at end of file
+}
+
+resource "cloudflare_record" "cloudflare_federation" {
```

Comment:
I would not pull Cloudflare into the individual cluster modules. They have no business knowing about "the outside world".

We already pass in the DNS name for the zones. We could actually create them in the main module and set the Cloudflare records there, only passing the zone's domain name into the cluster and having each cluster use a data source to get access to the whole zone instead. That would be cleaner, as we don't have to pass around Cloudflare credentials into each and every workspace.